internal/core/src/common/QueryResult.h

```diff
@@ -45,6 +146,37 @@ struct SearchResult {
         return topk_per_nq_prefix_sum_[total_nq_];
     }
 
+ public:
```

Comment:
for every query_vector, we need to assemble a VectorIterator which contains several Knowhere_iterators.
For example, if input query_vectors is 10 and 10 chunks totally.
Then, there will be 10 VectorIterators created accordingly and each of them integrates 10 Knowhere iterators.
Internally, I use priority queue to solve the sort problem inside the vectorIterator